ironmon_tracker/CustomCode.lua

```diff
@@ -458,6 +467,50 @@ function CustomCode.logError(err)
 	end
 end
 
+---Properly executes each extension's startup function, respecting any extension dependencies (cascade loading)
+function CustomCode.executeExtensionStartups()
+	if not CustomCode.isEnabled() then return end
+
+	local startedExtensionKeys = {}
+
+	local _tryStartupExtensions = function(extensionsToStartup)
+		local anyNewStarted = false
+		local skippedExtensions = {}
+		for _, ext in pairs(extensionsToStartup or {}) do
+			local allRequiredAreStarted = true
+			for _, requiredKey in pairs(ext.selfObject and ext.selfObject.requiredExtKeys or {}) do
+				if not Utils.isNilOrEmpty(requiredKey) and not startedExtensionKeys[requiredKey] then
+					allRequiredAreStarted = false
+					break
+				end
+			end
+			if allRequiredAreStarted then
+				startedExtensionKeys[ext.key] = true
+				anyNewStarted = true
+				-- Run its startup function
+				local functToExec = ext.selfObject and ext.selfObject["startup"]
+				if type(functToExec) == "function" then
+					CustomCode.tryExecute(ext.selfObject.name, "startup", functToExec)
+				end
+			else
+				table.insert(skippedExtensions, ext)
+			end
+		end
+		return anyNewStarted, skippedExtensions
+	end
+
+	-- Keep trying to startup extensions in a cascading fashion, until no new ones get started
+	local anyNewStarted = false
+	local unstartedExtensions = CustomCode.EnabledExtensions
+	for _ = 1, 999, 1 do
+		anyNewStarted, unstartedExtensions = _tryStartupExtensions(unstartedExtensions)
+		-- No new extensions were started, no need to keep trying
+		if not anyNewStarted or #unstartedExtensions == 0 then
+			break
+		end
+	end
```

Comment:
a while loop feels like it'd be more suitable here, aside from the above issue

Reply:
i purposely want to avoid a `while` loop whenever possible, since if we accidentally miss on the condition to exit, it will freeze the emulator. thus, a `for` loop with just a large iteration cap is generally safer for emulator